File: src/easy/two-sum/1.js
/*
 * @lc app=leetcode id=1 lang=javascript
 *
 * [1] Two Sum
 * Approaching 1
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  const recordMap = new Map();
  for (let i = 0; i < nums.length; i++) {
    if (recordMap.has(nums[i])) {
      return [recordMap.get(nums[i]), i];
    } else {
      recordMap.set(target - nums[i], i);
      continue;
    }
  }
};
// @lc code=end

File: src/medium/two-sum-ii-input-array-is-sorted/1.go
package main

import "fmt"

/*
 * @lc app=leetcode id=167 lang=golang
 *
 * [167] Two Sum II - Input Array Is Sorted
 */

func main() {
	fmt.Println(twoSum([]int{5, 25, 75}, 100))
}

// @lc code=start
func twoSum(numbers []int, target int) []int {
	l, r := 0, len(numbers)-1
	for l < len(numbers)-1 {
		if target > (numbers[l] + numbers[r]) {
			l++
		} else if target < (numbers[l] + numbers[r]) {
			r--
		} else {
			return []int{l + 1, r + 1}
		}
	}
	return []int{}
}

// @lc code=end
